src/Orchard.Web/Modules/Orchard.Alias/Controllers/AdminController.cs

```diff
@@ -105,8 +98,48 @@ public class AdminController : Controller {
                 default:
                     throw new ArgumentOutOfRangeException();
             }
+            return RedirectToAction("IndexUnmanaged");
+        }
+
+        public ActionResult IndexManaged(AdminIndexOptions options, PagerParameters pagerParameters) {
+            if (!Services.Authorizer.Authorize(StandardPermissions.SiteOwner, T("Not authorized to manage aliases")))
+                return new HttpUnauthorizedResult();
+
+            var pager = new Pager(Services.WorkContext.CurrentSite, pagerParameters);
+
+            // default options
+            if (options == null)
+                options = new AdminIndexOptions();
+
+
+            var aliases = _aliasHolder.GetMaps().SelectMany(x => x.GetAliases()).Where(x => x.IsManaged);
+
+            if (!String.IsNullOrWhiteSpace(options.Search)) {
+                var invariantSearch = options.Search.ToLowerInvariant();
+                aliases = aliases.Where(x => x.Path.ToLowerInvariant().Contains(invariantSearch));
+            }
 
-            return RedirectToAction("Index");
+            aliases = aliases.ToList();

```

Comment:
No, since you're potentially doing a sort on it on line 128, which should happen server side in the DB layer. You're already doing a ToList on it on line 139.